app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -61,18 +62,27 @@ def collect_disks
     disks = hardware.disks.map { |disk| "#{disk.storage.ems_ref}/disks/#{disk.filename}" }
     vm_disks = []
 
-    ext_management_system.with_provider_connection do |rhevm|
-      disks.each do |disk|
-        begin
-          vm_disks << Ovirt::Disk.find_by_href(rhevm, disk)
-        rescue Ovirt::MissingResourceError
-          nil
+    unless ext_management_system.nil?
+      ext_management_system.with_provider_connection do |rhevm|
+        disks.each do |disk|
+          begin
+            vm_disks << Ovirt::Disk.find_by_href(rhevm, disk)
+          rescue Ovirt::MissingResourceError
+            nil
+          end
         end
       end
     end
+
     vm_disks
   end
 
+  def disconnect_inv
```

Comment:
This method is used to remove ext_management_system from a vm when vm is removed by `save_inventory`. In the automation I noticed that we stop and remove vm very fast which causes stop refresh to delete it (ext_management_system = nil). We need to make sure we will be able to run disconnect_storage before ems is removed. I talked to @agrare about it and he suggested this approach.